eip-0043.md

```diff
@@ -0,0 +1,129 @@
+both the Context data _and_ the private keys, which are stored on the Cold Wallet device,
+and so the Prover must run on the Cold Wallet device.
+
+And that is the problem, we cannot transfer unsigned transaction along with all the
+contexts for each input to the Cold Wallet via QR codes. QR codes have limit of 4K bytes
+on the maximum size of serialized data. Even simplest transactions when serialized with
+required Contexts will exceed this limit.
+
+## Reduced Transaction data
+
+Here we introduce a new data structure and serialization format called
+`ReducedTransaction`.
+
+```
+ReducedTransaction:
+  - unsignedTx: UnsignedTransaction
+  - reducedInputs: Seq[ReducedInputData]
+  - txCost: Int
+
+UnsignedTransaction:
+  - inputs: Seq[UnsignedInput],
+  - dataInputs: Seq[DataInput],
+  - outputCandidates: Seq[ErgoBoxCandidate]
+
+UnsignedInput:
+  - boxId: BoxId
+  - extension: ContextExtension
+
+ReducedInputData:
+- reductionResult: ReductionResult
+
+ReductionResult:
+  - value: SigmaBoolean
+  - cost: Long
+```
+
+Thus, the `ReducedTransaction` instance contains unsigned transaction augmented with
+one `ReductionResult` for each `UnsignedInput`.
+
+Note that `UnsignedInput` object doesn't contain `ergoTree`, `additionalTokens`,
+`additionalRegisters` and other properties of
+[ErgoBox](https://github.com/ScorexFoundation/sigmastate-interpreter/blob/4533b6a7ae86ada20f3136c70a67a920ae7c43e1/sigmastate/src/main/scala/org/ergoplatform/ErgoBox.scala#L51)
+which are necessary to perform
+[ErgoTree](https://github.com/ScorexFoundation/sigmastate-interpreter/blob/1a1b003bc30e490d8b5af30e7670227e54e682c2/sigmastate/src/main/scala/sigmastate/Values.scala#L1014)
+reduction and which are part of the
+[Context](https://github.com/ScorexFoundation/sigmastate-interpreter/blob/e5127f6743db824f7280881cd5c4ecd336075e2f/sigmastate/src/main/scala/org/ergoplatform/ErgoLikeContext.scala#L51)
+data structure required by the
+[prove](https://github.com/ScorexFoundation/sigmastate-interpreter/blob/f24833d8d4572d77e4a93e5b69360335cb2d7dc1/sigmastate/src/main/scala/sigmastate/interpreter/ProverInterpreter.scala#L104)
+method to generate a proof (aka signature).
+
+This is because those context data is not required to generate proof
+once ErgoTree is reduced to ReductionResult containing sigma proposition.
+
+## Data Formats
+
+### ReducedTransaction
+
+ReducedTransaction is serialized to bytes array in the format described below. Note, field
+names are not serialized.
+
+| Field Name      | Format              | Description                                       |
+|-----------------|---------------------|---------------------------------------------------|
+| `messageSize`   | `VLQ(UInt)`         | Number of bytes in the messageToSign of the transaction |
+| `messageToSign` | `Bytes`             | serialized bytes of the unsignedTx.messageToSign  |
+| `reducedInputs` | `ReductionResult*` | serialized reduced inputs                         |
```

Comment:
```suggestion
| `reducedInputs` | `ReductionResult*` | serialized reduced inputs                         |
| `txCost`        | `VLQ(UInt)`        | transaction cost according to the prover          |
```

Reply:
We have the tx cost also serialized as part of ReducedTransaction in sigma-rust since the early days.

Reply:
@greenhat, just checked in the git history, there has never been txCost in Appkit's reference implementation. It doesn't make sense to store it, because there is are costs for ReducedResults which can be summed up.
There is probably a bug in sigma-rust. Please check how it is used in practice and does it really make sense to store it.

Reply:
Thats weird. I added it after our discussion here - https://discord.com/channels/668903786361651200/860527298292613151/912658058973437993

Reply:
@MrStahlfelge, I think the resolution largely depends on how this is implemented in ErgoPay and Mobile wallet, we cannot break all the dapps. Does it have txCost at the end of ReducedTransaction?

Reply:
Yes, Appkit saves the cost at the end:

https://github.com/ergoplatform/ergo-appkit/blob/5f15c28ad61ccef9dcd9cf6ab1407dcaa20539b2/lib-impl/src/main/java/org/ergoplatform/appkit/impl/ReducedTransactionImpl.java#L74

https://github.com/ergoplatform/ergo-appkit/blob/2bb32fb0db30381266865b19f96148dd5537744b/lib-impl/src/main/java/org/ergoplatform/appkit/impl/BlockchainContextBase.java#L49

Reply:
Ok, I moved cost saving to the serializer here https://github.com/ergoplatform/ergo-appkit/pull/217